### PR TITLE
Reallow single quote in file names

### PR DESCRIPTION
**Pull request by PVince81**
Regression from stable8, introduced by https://github.com/owncloud/core/commit/2f18a09a20cf62438c742c3020215cab03bc997d

This PR adds a unit test (in the spirit of test first).
Need to look into a solution.

Currently the sanitization block converts "a'b" to "a&#39b" here https://github.com/owncloud/core/commit/2f18a09a20cf62438c742c3020215cab03bc997d#diff-43d02ea7c0da1686b8ca6e343392cff1R508 which fails the final comparison.
- [x] fix the actual bug
- [x] add unit test

@LukasReschke @DeepDiver1975

### CODE REVIEW

**Comment by LukasReschke** [comment]
I'm taking this one.

**Comment by karlitschek** [comment]
:+1:

**Comment by LukasReschke** [comment]
Okay… Got the bug…The PHP documentation is misleading here as it indicates that `FILTER_SANITIZE_STRING` in combination with a `STRIP` flag will only strip characters that are specified within the stripping flag. However, it actually strips regular tags as well…

I changed it to `FILTER_UNSAFE_RAW` which should indeed do the work. – Tests should pass now.

**Comment by DeepDiver1975** [comment]
:+1: thx for the tests

File: tests/lib/files/pathverificationtest.php [hunk 1]
```diff
@@ -230,4 +230,30 @@ public function providesReservedNamesWindows() {
...
+	public function testPathVerificationValidPaths($fileName) {
+		$storage = new Local(['datadir' => '']);
+
+		\Test_Helper::invokePrivate($storage, 'verifyPosixPath', [$fileName]);
+		\Test_Helper::invokePrivate($storage, 'verifyWindowsPath', [$fileName]);
+		// nothing thrown
+		$this->assertTrue(true);
+	}
+
+	public function providesValidPosixPaths() {
```
Comment (nickvergessen) [inline tests/lib/files/pathverificationtest.php]: Would be nice to have a Invalid test suite aswell.
If that throws an Exception, use `@expectedException` https://phpunit.de/manual/current/en/writing-tests-for-phpunit.html#writing-tests-for-phpunit.exceptions

Comment (LukasReschke) [inline tests/lib/files/pathverificationtest.php]: Line 90 and following :see_no_evil:

Comment (nickvergessen) [inline tests/lib/files/pathverificationtest.php]: Okay, they are already in place above, sorry

Comment (PVince81) [inline tests/lib/files/pathverificationtest.php]: Yeah, let me reformulate it the other way for everyone: when write unit tests, please don't forget the "positive" tests too :smile:

**Pull request merged**